File: add/codesnippet/CSharp/t-system.web.ui.ihierarc_1_1.cs
    using System;
    using System.Collections;
    using System.IO;
    using System.Runtime.InteropServices;
    using System.Security.Permissions;
    using System.Web;
    using System.Web.UI;
    using System.Web.UI.WebControls;

    public class FileSystemDataSource :
        HierarchicalDataSourceControl, IHierarchicalDataSource
    {
        private FileSystemDataSourceView view = null;

        public FileSystemDataSource() : base() { }

        protected override HierarchicalDataSourceView
            GetHierarchicalView(string viewPath)
        {
            view = new FileSystemDataSourceView(viewPath);
            return view;
        }
    }

File: add/codesnippet/VisualBasic/t-system.web.ui.ihierarc_1_1.vb
Imports System
Imports System.Collections
Imports System.IO
Imports System.Runtime.InteropServices
Imports System.Security.Permissions
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Namespace Samples.AspNet

    Public Class FileSystemDataSource
        Inherits HierarchicalDataSourceControl

        Public Sub New()
        End Sub

        Private view As FileSystemDataSourceView = Nothing

        Protected Overrides Function GetHierarchicalView( _
            ByVal viewPath As String) As HierarchicalDataSourceView

            view = New FileSystemDataSourceView(viewPath)
            Return view
        End Function

    End Class